lib/src/main/kotlin/com/swmansion/starknet/data/types/Responses.kt

```diff
@@ -144,22 +148,33 @@ sealed class Syncing : StarknetResponse {
     abstract val highestBlockNumber: Int
 }
 
-@Serializable
-data class NotSyncingResponse(
+@Suppress("DataClassPrivateConstructor")
+@Serializable(with = NotSyncingResponseSerializer::class)
+data class NotSyncingResponse private constructor(
     override val status: Boolean,
 
-    override val startingBlockHash: Felt = Felt.ZERO,
+    override val startingBlockHash: Felt,
 
-    override val startingBlockNumber: Int = 0,
+    override val startingBlockNumber: Int,
 
-    override val currentBlockHash: Felt = Felt.ZERO,
+    override val currentBlockHash: Felt,
 
-    override val currentBlockNumber: Int = 0,
+    override val currentBlockNumber: Int,
 
-    override val highestBlockHash: Felt = Felt.ZERO,
+    override val highestBlockHash: Felt,
 
-    override val highestBlockNumber: Int = 0,
-) : Syncing()
+    override val highestBlockNumber: Int,
+) : Syncing() {
+    constructor(status: Boolean) : this(
+        status,
+        Felt.ZERO,
+        0,
+        Felt.ZERO,
+        0,
+        Felt.ZERO,
+        0,
```

Comment:
Maybe we should use named arguments in these constructors for readability + to ensure nothing breaks upon changing values in dataclasses?

Reply:
This applies to other secondary constructors